AuditModule needs the SecurityService, which is not available with disabled security

Hi @tchiotludo @ebrard, it seems like the Audit-Feature does not support disabled security:

```Message: No bean of type [io.micronaut.security.utils.SecurityService] exists. The bean [SecurityService] is disabled because it is within the package [io.micronaut.security] which is disabled due to bean requirements: 
* Property [micronaut.security.enabled] with value [false] should not equal: false

Path Taken: new TopicController() --> TopicController.kafkaWrapper --> $KafkaWrapperRequestScope$Definition$Intercepted.auditModule --> AuditModule.securityService
        at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2181)
        at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForField(AbstractInitializableBeanDefinition.java:1672)
        at org.akhq.modules.$AuditModule$Definition.inject(Unknown Source)
        at org.akhq.modules.$AuditModule$Definition.instantiate(Unknown Source)
        at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2309)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2279)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2291)
        at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3054)
        at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
        at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2956)
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2917)
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2730)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1729)
        at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:89)
        at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2165)
        at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForField(AbstractInitializableBeanDefinition.java:1672)
        at org.akhq.modules.$KafkaWrapperRequestScope$Definition.inject(Unknown Source)
        at org.akhq.modules.$KafkaWrapperRequestScope$Definition$Intercepted$Definition.instantiate(Unknown Source)
        at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2309)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2279)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2291)
        at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3054)
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2928)
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2730)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1729)
        at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:89)
        at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2165)
        at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForField(AbstractInitializableBeanDefinition.java:1672)
        at org.akhq.controllers.$TopicController$Definition.inject(Unknown Source)
        at org.akhq.controllers.$TopicController$Definition.instantiate(Unknown Source)
        at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2309)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2279)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2291)
        at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3054)
        at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
        at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2956)
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2917)
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2891)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1695)
        at io.micronaut.context.DefaultBeanContext$BeanContextExecutionHandle.getTarget(DefaultBeanContext.java:4247)
        at io.micronaut.context.DefaultBeanContext$BeanContextUnsafeExecutionHandle.invokeUnsafe(DefaultBeanContext.java:4214)
        at io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:263)
        at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:74)
        at io.micronaut.http.server.RouteExecutor.executeRouteAndConvertBody(RouteExecutor.java:480)
        at io.micronaut.http.server.RouteExecutor.lambda$callRoute$6(RouteExecutor.java:457)
        at io.micronaut.core.execution.ExecutionFlow.lambda$async$1(ExecutionFlow.java:87)
        at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:141)
        at io.micrometer.core.instrument.Timer.lambda$wrap$0(Timer.java:196)
        at io.micronaut.core.propagation.PropagatedContext.lambda$wrap$3(PropagatedContext.java:211)
        at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:141)
        at io.micrometer.core.instrument.Timer.lambda$wrap$0(Timer.java:196)
        at io.micronaut.core.propagation.PropagatedContext.lambda$wrap$3(PropagatedContext.java:211)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [io.micronaut.security.utils.SecurityService] exists. The bean [SecurityService] is disabled because it is within the package [io.micronaut.security] which is disabled due to bean requirements: 
* Property [micronaut.security.enabled] with value [false] should not equal: false

        at io.micronaut.context.DefaultBeanContext.newNoSuchBeanException(DefaultBeanContext.java:2766)
        at io.micronaut.context.DefaultApplicationContext.newNoSuchBeanException(DefaultApplicationContext.java:304)
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2735)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1729)
        at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:89)
        at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2165)
        ... 54 common frames omitted
```

Would you please be so kind to find a way to support it soon?

--- a/src/main/java/org/akhq/modules/AuditModule.java
+++ b/src/main/java/org/akhq/modules/AuditModule.java
@@ -22,7 +22,8 @@ import java.util.Optional;
 @Singleton
 @Slf4j
 public class AuditModule {
-    SecurityService securityService;
+    @Inject
+    Optional<SecurityService> securityService;
 
     @Inject
     KafkaModule kafkaModule;
@@ -30,20 +31,9 @@ public class AuditModule {
     @Inject
     Audit auditConfig;
 
-    @Value("${micronaut.security.enabled}")
-    Boolean securityEnabled;
-
     @Inject
     ApplicationContext applicationContext;
 
-    @PostConstruct
-    public void init() {
-        if (securityEnabled) {
-            securityService = applicationContext.getBean(SecurityService.class);
-        } else {
-            securityService = new NoOpSecurityService();
-        }
-    }
 
     private final ObjectMapper mapper = new ObjectMapper();
 
@@ -58,7 +48,7 @@ public class AuditModule {
 
 
         byte[] value;
-        securityService.username().ifPresent(event::setUserName);
+        securityService.flatMap(SecurityService::username).ifPresent(event::setUserName);
         try {
             value = mapper.writeValueAsBytes(event);
         } catch (Exception e) {
@@ -71,29 +61,6 @@ public class AuditModule {
                 log.error("Audit data cannot be sent to Kafka", exception);
             }
         });
-    }
-
-    private static class NoOpSecurityService implements SecurityService {
-
-        @Override
-        public Optional<String> username() {
-            return Optional.empty();
-        }
-
-        @Override
-        public Optional<Authentication> getAuthentication() {
-            return Optional.empty();
-        }
-
-        @Override
-        public boolean isAuthenticated() {
-            return false;
-        }
-
-        @Override
-        public boolean hasRole(String role) {
-            return false;
-        }
     }
 
 }

--- a/src/main/java/org/akhq/modules/AuditModule.java
+++ b/src/main/java/org/akhq/modules/AuditModule.java
@@ -1,7 +1,11 @@
 package org.akhq.modules;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.context.ApplicationContext;
 import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.security.authentication.Authentication;
+import jakarta.annotation.PostConstruct;
 import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
 import lombok.extern.slf4j.Slf4j;
@@ -13,12 +17,11 @@ import org.apache.kafka.clients.producer.ProducerRecord;
 import org.apache.kafka.clients.producer.RecordMetadata;
 
 import java.time.LocalDateTime;
+import java.util.Optional;
 
 @Singleton
 @Slf4j
 public class AuditModule {
-
-    @Inject
     SecurityService securityService;
 
     @Inject
@@ -26,6 +29,21 @@ public class AuditModule {
 
     @Inject
     Audit auditConfig;
+
+    @Value("${micronaut.security.enabled}")
+    Boolean securityEnabled;
+
+    @Inject
+    ApplicationContext applicationContext;
+
+    @PostConstruct
+    public void init() {
+        if (securityEnabled) {
+            securityService = applicationContext.getBean(SecurityService.class);
+        } else {
+            securityService = new NoOpSecurityService();
+        }
+    }
 
     private final ObjectMapper mapper = new ObjectMapper();
 
@@ -53,6 +71,29 @@ public class AuditModule {
                 log.error("Audit data cannot be sent to Kafka", exception);
             }
         });
+    }
+
+    private static class NoOpSecurityService implements SecurityService {
+
+        @Override
+        public Optional<String> username() {
+            return Optional.empty();
+        }
+
+        @Override
+        public Optional<Authentication> getAuthentication() {
+            return Optional.empty();
+        }
+
+        @Override
+        public boolean isAuthenticated() {
+            return false;
+        }
+
+        @Override
+        public boolean hasRole(String role) {
+            return false;
+        }
     }
 
 }

--- a/src/main/java/org/akhq/modules/AuditModule.java
+++ b/src/main/java/org/akhq/modules/AuditModule.java
@@ -1,22 +1,14 @@
 package org.akhq.modules;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import io.micronaut.context.ApplicationContext;
-import io.micronaut.context.annotation.Requires;
-import io.micronaut.context.annotation.Value;
-import io.micronaut.security.authentication.Authentication;
-import jakarta.annotation.PostConstruct;
+import io.micronaut.security.utils.SecurityService;
 import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
 import lombok.extern.slf4j.Slf4j;
 import org.akhq.configs.Audit;
 import org.akhq.models.audit.AuditEvent;
-import io.micronaut.security.utils.SecurityService;
-import org.apache.kafka.clients.producer.Callback;
 import org.apache.kafka.clients.producer.ProducerRecord;
-import org.apache.kafka.clients.producer.RecordMetadata;
 
-import java.time.LocalDateTime;
 import java.util.Optional;
 
 @Singleton
@@ -30,10 +22,6 @@ public class AuditModule {
 
     @Inject
     Audit auditConfig;
-
-    @Inject
-    ApplicationContext applicationContext;
-
 
     private final ObjectMapper mapper = new ObjectMapper();
 

--- a/src/test/java/org/akhq/modules/audit/KafkaAuditModuleNoSecTest.java
+++ b/src/test/java/org/akhq/modules/audit/KafkaAuditModuleNoSecTest.java
@@ -1,0 +1,44 @@
+package org.akhq.modules.audit;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.akhq.KafkaClusterExtension;
+import org.akhq.KafkaTestCluster;
+import org.akhq.repositories.TopicRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * This only tests instantiation
+ */
+@ExtendWith(KafkaClusterExtension.class)
+@MicronautTest(environments = "audit-no-security")
+public class KafkaAuditModuleNoSecTest {
+
+    @Inject
+    @InjectMocks
+    protected TopicRepository topicRepository;
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void before() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void topicAudit() throws ExecutionException, InterruptedException, IOException {
+        topicRepository.create(KafkaTestCluster.CLUSTER_ID, "dummy-no-sec", 1, (short) 1, Collections.emptyList()
+        );
+
+    }
+
+}

--- a/src/test/java/org/akhq/modules/audit/KafkaAuditModuleNoSecTest.java
+++ b/src/test/java/org/akhq/modules/audit/KafkaAuditModuleNoSecTest.java
@@ -40,7 +40,7 @@ public class KafkaAuditModuleNoSecTest extends AbstractTest  {
 
     @Test
     void topicAuditNoSec() throws ExecutionException, InterruptedException {
-        topicRepository.create(KafkaTestCluster.CLUSTER_ID, "dummy-no-sec", 1, (short) 1, Collections.emptyList()
+        topicRepository.create(KafkaTestCluster.CLUSTER_ID, "generated_dummy-no-sec", 1, (short) 1, Collections.emptyList()
         );
     }
 

--- a/src/test/java/org/akhq/modules/audit/KafkaAuditModuleNoSecTest.java
+++ b/src/test/java/org/akhq/modules/audit/KafkaAuditModuleNoSecTest.java
@@ -1,15 +1,18 @@
 package org.akhq.modules.audit;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.context.ApplicationContext;
 import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
 import jakarta.inject.Inject;
+import org.akhq.AbstractTest;
 import org.akhq.KafkaClusterExtension;
 import org.akhq.KafkaTestCluster;
+import org.akhq.modules.KafkaModule;
 import org.akhq.repositories.TopicRepository;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 
 import java.io.IOException;
@@ -21,13 +24,14 @@ import java.util.concurrent.ExecutionException;
  */
 @ExtendWith(KafkaClusterExtension.class)
 @MicronautTest(environments = "audit-no-security")
-public class KafkaAuditModuleNoSecTest {
+public class KafkaAuditModuleNoSecTest extends AbstractTest  {
 
     @Inject
     @InjectMocks
     protected TopicRepository topicRepository;
 
-    private final ObjectMapper mapper = new ObjectMapper();
+    @Mock
+    ApplicationContext applicationContext;
 
     @BeforeEach
     void before() {
@@ -35,10 +39,9 @@ public class KafkaAuditModuleNoSecTest {
     }
 
     @Test
-    void topicAudit() throws ExecutionException, InterruptedException, IOException {
+    void topicAuditNoSec() throws ExecutionException, InterruptedException {
         topicRepository.create(KafkaTestCluster.CLUSTER_ID, "dummy-no-sec", 1, (short) 1, Collections.emptyList()
         );
-
     }
 
 }

--- a/src/test/resources/application-audit-no-security.yml
+++ b/src/test/resources/application-audit-no-security.yml
@@ -1,6 +1,6 @@
 micronaut:
   security:
-    enabled: true
+    enabled: false
 akhq:
   audit:
     enabled: true

--- a/src/test/resources/application-audit-no-security.yml
+++ b/src/test/resources/application-audit-no-security.yml
@@ -1,6 +1,6 @@
 micronaut:
   security:
-    enabled: false
+    enabled: true
 akhq:
   audit:
     enabled: true

--- a/src/test/resources/application-audit-no-security.yml
+++ b/src/test/resources/application-audit-no-security.yml
@@ -1,0 +1,8 @@
+micronaut:
+  security:
+    enabled: false
+akhq:
+  audit:
+    enabled: true
+    cluster-id: test
+    topic-name: audit

